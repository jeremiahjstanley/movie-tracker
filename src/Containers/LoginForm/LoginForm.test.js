import React from 'react';
import { LoginForm, mapStateToProps, mapDispatchToProps } from './index.js';
import { shallow, mount } from 'enzyme';
import { BrowserRouter, Link, withRouter } from 'react-router-dom';


describe('Login Form tests', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<LoginForm />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state of email, password and error message, all empty strings', () => {
    const wrapper = shallow(<LoginForm />);
    const expected = {
      email: '',
      password: '',
      errorMessage: ''
    };
    expect(wrapper.state()).toEqual(expected);
  });

  it('should update the state when handleChange is invoked', () => {
    const wrapper = shallow(<LoginForm />);

    const mockEvent = { target: { value: 'NickCage@aol.com', name: 'email' } };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('email')).toBe('NickCage@aol.com');
  });

  it('should invoke handleChange when the email field changes', () => {
    const wrapper = mount(withRouter(<LoginForm />), { disableLifecycleMethods: true });
    const spy = spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    const mockEvent = { target: { value: 'NickCage@aol.com', name: 'email' } };
    wrapper.find('.email-input').simulate('change', mockEvent);

    expect(spy).toHaveBeenCalledWith();
  });

  

  // it('should calls handleChange when the password field changes', () => {
  //   const wrapper = shallow(<LoginForm />);
  //   const spy = spyOn(wrapper.instance(), 'handleChange');
  //   wrapper.instance().forceUpdate();
  //   const mockEvent = { target: { value: 'password', name: 'password' } };
  //   wrapper.find('.password-input').simulate('change', mockEvent);

  //   expect(spy).toHaveBeenCalled();
  // });

  // it('should reset state after handle submit is called', () => {
    
  //   expect('').toEqual('');
  // });


  describe('mapStateToProps', () => {
    it('should return an object with the email, name and favorites array', () => {
      const mockState = { "favorites": [{ "title": "Con Air" }], "login": { "email": "nickcage@aol.com", "name": "Nick Cage" } };
      const expected = { "email": "nickcage@aol.com", "favorites": [{ "title": "Con Air" }], "name": "Nick Cage" };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

  });

});