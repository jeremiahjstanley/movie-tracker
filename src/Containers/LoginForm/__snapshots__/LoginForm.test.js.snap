// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="exit-form"
          replace={false}
          to="/"
>
          X
</Link>,
        <h1
          className="form-header"
>
          Sign In
</h1>,
        <input
          aria-label="Please Enter Your Email"
          className="email-field"
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="text"
          value=""
/>,
        <input
          aria-label="Please Enter Your Password"
          className="password-input"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button
          aria-label="Sign in to your account"
>
          Login
</button>,
        <h3>
           
          
           
</h3>,
        <Link
          replace={false}
          to="/signup"
>
          Don't have an account?
</Link>,
      ],
      "className": "form login-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "X",
          "className": "exit-form",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "X",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
          "className": "form-header",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Please Enter Your Email",
          "className": "email-field",
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Please Enter Your Password",
          "className": "password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Sign in to your account",
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "",
          " ",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Don't have an account?",
          "replace": false,
          "to": "/signup",
        },
        "ref": null,
        "rendered": "Don't have an account?",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="exit-form"
            replace={false}
            to="/"
>
            X
</Link>,
          <h1
            className="form-header"
>
            Sign In
</h1>,
          <input
            aria-label="Please Enter Your Email"
            className="email-field"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
/>,
          <input
            aria-label="Please Enter Your Password"
            className="password-input"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            aria-label="Sign in to your account"
>
            Login
</button>,
          <h3>
             
            
             
</h3>,
          <Link
            replace={false}
            to="/signup"
>
            Don't have an account?
</Link>,
        ],
        "className": "form login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "X",
            "className": "exit-form",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "X",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "className": "form-header",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Please Enter Your Email",
            "className": "email-field",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Please Enter Your Password",
            "className": "password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Sign in to your account",
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "",
            " ",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Don't have an account?",
            "replace": false,
            "to": "/signup",
          },
          "ref": null,
          "rendered": "Don't have an account?",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
